app/src/main/java/io/github/droidkaigi/confsched/fragment/SessionDetailFragment.java

```diff
@@ -64,14 +65,35 @@ public void onCreate(@Nullable Bundle savedInstanceState) {
         AppUtil.setTaskDescription(activity, session.title, ContextCompat.getColor(activity, session.category.getVividColorResId()));
     }
 
+    // Receive all viewModel's properties changed.
+    private final Observable.OnPropertyChangedCallback onPropertyChanged = new Observable.OnPropertyChangedCallback() {
```

Comment:
> In origin, the problem is that I have to call setResult() method is SessionDetailFragment. 
> This is because I want to refresh the session checked state in sessions list (Previous Activity).

I don't use ``onActivityResult``.
View just only subscribe model via viewmodel.

I drew simple diagram.

![droidkaigi_mvvm_01](https://cloud.githubusercontent.com/assets/401369/16893369/adbffa58-4b70-11e6-8e69-1ab2cdfc9c48.png)

* ``SessionsModel`` is almost singleton (or same as app lifecycle). I agreed @ninjinkun said ["App is big singleton"](http://ninjinkun.hatenablog.com/entry/2016/02/02/183136).
* ``SessionDetailFragment`` subscribe ``SessionModel`` properties changes via ``SessionDetailViewModel``.
* ``SessionsFragment`` subscribe ``SessionsModel`` items changes via ``SessionsViewModel``.
* Also ``OtherFragment`` subscribe ``SessionModel(or SessionsModel)`` via ``OtherViewModel``.

In this way, if Model changes from some action, then all views receive "onChanged" and refresh automatically.

**It's Reactive world!** 👍